docs/source/torch.compiler_api.rst

```diff
@@ -25,3 +25,7 @@ For a quick overview of ``torch.compiler``, see :ref:`torch.compiler_overview`.
      is_compiling
      is_dynamo_compiling
      is_exporting
+     skip_guard_on_inbuilt_nn_modules_unsafe
+     skip_guard_on_all_nn_modules_unsafe
+     keep_tensor_guards
+     skip_guard_on_globals
```

Comment:
I think the conclusions from the meeting on friday were:
1) don't expose custom guards yet,
2) Instead we are going to expose a long list of things that people want and then look at the user feedback

so I think what you have now is aligned with that discussion

Reply:
Yes, I will add a keep_guard_on_inputs. I am not too excited about require_grad_mode and require_inference_mode. I think we should just fix the requires_inference_mode (and provide one option that is more compatible with compile). For requires_grad, we should tell the user to perhaps use `torch.no_grad`.